### PR TITLE
Add more weirdness to weird-exprs.rs

### PR DESCRIPTION
**Pull request by oberien**

### CODE REVIEW

File: tests/ui/weird-exprs.rs [hunk 1]
```diff
@@ -231,6 +231,31 @@ fn infcx() {
...
 }
 
+fn return_already() -> impl std::fmt::Debug {
+    loop {
+        return !!!!!!!
+        break !!!!!!1111
+    }
+}
+
+fn cursed_macros() -> impl std::fmt::Debug {
```
Comment (bjorn3) [inline tests/ui/weird-exprs.rs:241]: This name is misleading. Those are not actually macro invocations, but keywords followed by `!` negations and regular blocks.

Comment (oberien) [reply tests/ui/weird-exprs.rs:241]: That's somewhat the whole purpose of `weird-exprs.rs`: Abuse rust's syntax to make an expression look like something different (ensuring that it'll always be parsed correctly). Similarly, `fn closure_matching()` does not in fact match on closures.

**Comment by cjgillot** [comment]
Could you elaborate on the purpose of this PR? Have those examples led to bugs? Or correspond to uncovered testing area? Or is it just for the fun of the weirdness?

**Comment by oberien** [comment]
I see `weird-exprs.rs` as a culmination of multiple "goals":
1. For weirdness sake. You should never encounter these examples in real code, but they are valid syntax. It's a fun game (and possibly check of one's knowledge about rust) to try and understand all the functions within the file.
2. Compiler testing. As those expressions likely aren't anywhere to be found in real code, they are unlikely to be tested. However, they abuse the syntax to possibly look like other valid rust code. As the file is part of the ui tests, adding more of these weird expressions ensures that the compiler will continue to parse them correctly.

**Comment by cjgillot** [comment]
r=me with @bjorn3 comment addressed

**Comment by oberien** [comment]
I've renamed the function to make it clear that these are in fact not real macros.

r? @cjgillot

**Pull request merged**